official/resnet/keras/keras_cifar_main.py

```diff
@@ -0,0 +1,185 @@
+  label = tf.sparse_to_dense(label, (cifar_main.NUM_CLASSES,), 1)
+  return image, label
+
+
+def run(flags_obj):
+  """Run ResNet Cifar-10 training and eval loop using native Keras APIs.
+
+  Args:
+    flags_obj: An object containing parsed flag values.
+
+  Raises:
+    ValueError: If fp16 is passed as it is not currently supported.
+  """
+  if flags_obj.enable_eager:
+    tf.enable_eager_execution()
+
+  dtype = flags_core.get_tf_dtype(flags_obj)
+  if dtype == 'fp16':
+    raise ValueError('dtype fp16 is not supported in Keras. Use the default '
+                     'value(fp32).')
+
+  per_device_batch_size = distribution_utils.per_device_batch_size(
+      flags_obj.batch_size, flags_core.get_num_gpus(flags_obj))
+
+  if flags_obj.use_synthetic_data:
+    input_fn = keras_common.get_synth_input_fn(
+        height=cifar_main.HEIGHT,
+        width=cifar_main.WIDTH,
+        num_channels=cifar_main.NUM_CHANNELS,
+        num_classes=cifar_main.NUM_CLASSES,
+        dtype=flags_core.get_tf_dtype(flags_obj))
+  else:
+    input_fn = cifar_main.input_fn
+
+  train_input_dataset = input_fn(
+      is_training=True,
+      data_dir=flags_obj.data_dir,
+      batch_size=per_device_batch_size,
+      num_epochs=flags_obj.train_epochs,
+      parse_record_fn=parse_record_keras)
+
+  eval_input_dataset = input_fn(
+      is_training=False,
+      data_dir=flags_obj.data_dir,
+      batch_size=per_device_batch_size,
+      num_epochs=flags_obj.train_epochs,
+      parse_record_fn=parse_record_keras)
+
+  optimizer = keras_common.get_optimizer()
+  strategy = distribution_utils.get_distribution_strategy(
+    flags_obj.num_gpus, flags_obj.turn_off_distribution_strategy)
+
+  model = resnet_cifar_model.resnet56(input_shape=(32, 32, 3),
+          classes=cifar_main.NUM_CLASSES)
+
+  model.compile(loss='categorical_crossentropy',
+                optimizer=optimizer,
+                metrics=['categorical_accuracy'],
+                distribute=strategy)
+
+  time_callback, tensorboard_callback, lr_callback = keras_common.get_callbacks(
+      learning_rate_schedule, cifar_main.NUM_IMAGES['train'])
+
+  train_steps = cifar_main.NUM_IMAGES['train'] // flags_obj.batch_size
+  train_epochs = flags_obj.train_epochs
+
+  if flags_obj.train_steps:
+    train_steps = min(flags_obj.train_steps, train_steps)
+    train_epochs = 1
+
+  num_eval_steps = (cifar_main.NUM_IMAGES['validation'] //
+                    flags_obj.batch_size)
+
+  validation_data = eval_input_dataset
+  if flags_obj.skip_eval:
+    num_eval_steps = None
+    validation_data = None
+
+  history = model.fit(train_input_dataset,
+      epochs=train_epochs,
+      steps_per_epoch=train_steps,
+      callbacks=[
+        time_callback,
+        lr_callback,
+        tensorboard_callback
+        ],
+      validation_steps=num_eval_steps,
+      validation_data=validation_data,
+      verbose=1)
+
+  if not flags_obj.skip_eval:
+    eval_output = model.evaluate(eval_input_dataset,
```

Comment:
what happened to the analyze results helper? we don't need that anymore, or to return these result from here? i thought the benchmark code depends on that?

Reply:
I fixed it handled sparse non sparse with unit tests.  I did not do imagenet but that can be done in another PR as I do not have a benchmark yet.
https://github.com/tensorflow/models/pull/5928/commits/61ec290757e495104f1150e00449187b765805c2